File: sources/julia/03_func_sum/main.jl
# task 03 func_sum — expected output: 100000000
# build: julia main.jl    run: julia main.jl

@noinline function add_one(n::Int64)::Int64
    return n + 1
end

function main()
    value = Int64(0)
    for _ in 1:100000000
        value = add_one(value)
    end
    println(value)
end

main()

File: sources/nim/03_func_sum/main.nim
# task 03 func_sum — expected output: 100000000
# build: nim c -d:release -o:prog main.nim    run: ./prog

proc addOne(n: int64): int64 {.noinline.} =
  n + 1

var value: int64 = 0
for _ in 0 ..< 100_000_000:
  value = addOne(value)
echo value
